### PR TITLE
 Add missing fields to WorkflowExecutionStartedEventAttributes

### LINKED ISSUE #425: Add missing fields to WorkflowExecutionStartedEventAttributes
WorkflowExecutionStartedEventAttributes misses the following fields that are present in SWF:

parentWorkflowExecution
parentInitiatedEventId
childPolicy
continuedExecutionRunId

### PR DESCRIPTION
**Pull request by samarabbas**
WorkflowExecutionStartedEventAttributes now also has support for parent
execution info, child policy and continued execution run id.

fixes #425

### CODE REVIEW

File: host/integration_test.go [hunk 1]
```diff
@@ -2642,9 +2642,12 @@ func (s *integrationSuite) TestContinueAsNewWorkflow() {
 	workflowComplete := false
 	continueAsNewCount := int32(10)
 	continueAsNewCounter := int32(0)
+	var previousRunID string
+	var lastRunStartedEvent *workflow.HistoryEvent
 	dtHandler := func(execution *workflow.WorkflowExecution, wt *workflow.WorkflowType,
 		previousStartedEventID, startedEventID int64, history *workflow.History) ([]byte, []*workflow.Decision, error) {
 		if continueAsNewCounter < continueAsNewCount {
+			previousRunID = execution.GetRunId()
```
Comment (wxing1292) [inline host/integration_test.go]: what does this previousRunID intend to do?

File: idl/github.com/uber/cadence/shared.thrift [hunk 2]
```diff
@@ -365,10 +365,15 @@ struct Decision {
...
 struct WorkflowExecutionStartedEventAttributes {
   10: optional WorkflowType workflowType
+  12: optional string parentWorkflowDomain
+  14: optional WorkflowExecution parentWorkflowExecution
+  16: optional i64 (js.type = "Long") parentInitiatedEventId
   20: optional TaskList taskList
   30: optional binary input
   40: optional i32 executionStartToCloseTimeoutSeconds
   50: optional i32 taskStartToCloseTimeoutSeconds
+  52: optional ChildPolicy childPolicy
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/shared.thrift]: if the "child policy" represents what this workflow's parent want this workflow to do, then we should use a better name

File: idl/github.com/uber/cadence/shared.thrift [hunk 3]
```diff
@@ -812,6 +817,7 @@ struct StartWorkflowExecutionRequest {
   80: optional string identity
   90: optional string requestId
   100: optional WorkflowIdReusePolicy workflowIdReusePolicy
+  110: optional ChildPolicy childPolicy
```
Comment (wxing1292) [inline idl/github.com/uber/cadence/shared.thrift]: same here

File: service/history/historyBuilder.go [hunk 4]
```diff
@@ -420,15 +420,24 @@ func (b *historyBuilder) addEventToHistory(event *workflow.HistoryEvent) *workfl
...
+	request := startRequest.StartRequest
 	historyEvent := b.msBuilder.createNewHistoryEvent(workflow.EventTypeWorkflowExecutionStarted)
 	attributes := &workflow.WorkflowExecutionStartedEventAttributes{}
 	attributes.WorkflowType = request.WorkflowType
 	attributes.TaskList = request.TaskList
 	attributes.Input = request.Input
 	attributes.ExecutionStartToCloseTimeoutSeconds = common.Int32Ptr(*request.ExecutionStartToCloseTimeoutSeconds)
 	attributes.TaskStartToCloseTimeoutSeconds = common.Int32Ptr(*request.TaskStartToCloseTimeoutSeconds)
+	attributes.ChildPolicy = request.ChildPolicy
+	attributes.ContinuedExecutionRunId = previousRunID
```
Comment (wxing1292) [inline service/history/historyBuilder.go]: previous run ID is better than ContinuedExecutionRunId, i assume

File: service/history/mutableStateBuilder.go [hunk 5]
```diff
@@ -875,7 +875,7 @@ func (e *mutableStateBuilder) AddWorkflowExecutionStartedEventForContinueAsNew(d
 
 	createRequest := &workflow.StartWorkflowExecutionRequest{
 		RequestId:                           common.StringPtr(uuid.New()),
-		Domain:                              common.StringPtr(previousExecutionState.executionInfo.DomainID),
+		Domain:                              common.StringPtr(domainName),
```
Comment (wxing1292) [inline service/history/mutableStateBuilder.go]: this is a bug fix?

File: idl/github.com/uber/cadence/shared.thrift (hunk 2 above)
Comment (samarabbas) [reply idl/github.com/uber/cadence/shared.thrift]: We are keeping our API as close to SWF as possible: https://docs.aws.amazon.com/amazonswf/latest/apireference/API_WorkflowExecutionStartedEventAttributes.html

File: idl/github.com/uber/cadence/shared.thrift (hunk 3 above)
Comment (samarabbas) [reply idl/github.com/uber/cadence/shared.thrift]: same as above.

File: service/history/historyBuilder.go (hunk 4 above)
Comment (samarabbas) [reply service/history/historyBuilder.go]: Same as above.

**Comment by johndpope** [comment]
this is a complete rip off of aws swf, right?

**Pull request merged**